src/host/selectionState.cpp

```diff
@@ -197,6 +197,37 @@ til::point Selection::GetSelectionAnchor() const noexcept
     return _coordSelectionAnchor;
 }
 
+// Routine Description:
+// - Gets the current selection begin and end anchor positions.
+// Arguments:
+// - ordered - if true, arrange the selection anchors such that the first anchor
+//             is at the top left corner, and the second one is at the bottom
+//             right of the selection area
+// Return Value:
+// - The current selection anchors
+std::pair<til::point, til::point> Selection::GetSelectionAnchors(const bool ordered) const noexcept
+{
+    if (!_fSelectionVisible)
+    {
+        return {};

```

Comment:
can we distinguish this from there being a single selection at 0,0? do we need to?

Reply:
For consistency, we should return something like `{ { 0, 0 }, { -1, -1 } }` to represent an empty selection. Fixed.

This also helped me find a bug in the possible range for CopyRequest's end coordinate. Since it's an _inclusive_ coordinate we should be clamping it within `{ buffer.width - 1, buffer.height - 1 }` ( and not `{ buffer.width, buffer.height }`). It is fixed too.

Thanks.